### ISSUE TITLE
dotnet vstest does not detect tests when running under a non-US locale

### ISSUE THREAD

**Issue by mark-raymond** [microsoft/vstest#2853]
## Description

`dotnet vstest` does not detect tests when running under a non-US locale. The is a problem that is present in Microsoft.NET.Test.Sdk 16.9.1 and 16.9.4, but was not present in 16.8.3.

## Steps to reproduce

- Clone [mark-raymond/dotnet-test-locale-test](https://github.com/mark-raymond/dotnet-test-locale-test)
- Run `./test.sh en_US` to see the test runner successfully run using the en_US locale
- Run `./test.sh <some other locale>` to see the test runner fail to find tests. At time of writing, I have tried en_GB, en_CA, and fr_FR, and all fail to find tests.

## Expected behavior

```
Microsoft (R) Test Execution Command Line Tool Version 16.9.4
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     1, Skipped:     0, Total:     1, Duration: 8 ms - /app/dotnet-test-locale-test.dll (net5.0)
```

## Actual behavior

```
Microsoft (R) Test Execution Command Line Tool Version 16.9.4
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
No test is available in /app/dotnet-test-locale-test.dll. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again.

Additionally, path to test adapters can be specified using /TestAdapterPath command. Example  /TestAdapterPath:<pathToCustomAdapters>.
```

## Diagnostic logs

[en_GB.host.21-04-14_13-32-40_29866_4.log](https://github.com/microsoft/vstest/files/6311138/en_GB.host.21-04-14_13-32-40_29866_4.log)
[en_GB.log](https://github.com/microsoft/vstest/files/6311140/en_GB.log)
[en_US.host.21-04-14_13-32-32_85293_4.log](https://github.com/microsoft/vstest/files/6311141/en_US.host.21-04-14_13-32-32_85293_4.log)
[en_US.log](https://github.com/microsoft/vstest/files/6311143/en_US.log)

## Environment

Originally noticed on an Ubuntu 18.04 build agent, and reproduced in an Ubuntu 20.04 Docker container.

[AB#1314786](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1314786)

**Comment by mark-raymond** [comment]
I've now bisected the NuGet preview releases, and this bug was introduced between 16.9.0-preview-20201020-06 and 16.9.0-preview-20201123-03.

**Comment by nohwnd** [comment]
Thank you for this repro. We are looking into it.

**Comment by Haplois** [comment]
[This workaround][1] can be used to fix this issue temporarily. We are looking into this issue.


[1]: https://github.com/microsoft/vstest/issues/2814#issuecomment-790244670

**Comment by Haplois** [comment]
This is fixed in #2860, and the fix is released on `v16.10.0-release-20210421-08` on our daily feeds.
